lib/classes/organisations_user_builder.rb

```diff
@@ -0,0 +1,59 @@
+class OrganisationsUserBuilder
+
+  # :organisations_user: {
+  #   :organisation_id,
+  #   :position,
+  #   user_attributes: {
+  #     :first_name,
+  #     :last_name,
+  #     :mobile,
+  #     :email
+  #   }
+
+  def initialize(params)
+    @organisation_id = params[:organisation_id].to_i
+    @user_attributes = params[:user_attributes]
+    @mobile = @user_attributes[:mobile]
+    @position = params[:position]
+    fail_with_error(I18n.t('organisations_user_builder.organisation.blank')) unless @organisation_id.present?
+    fail_with_error(I18n.t('organisations_user_builder.user.mobile.blank')) unless @mobile.present?
+  end
+
+  def build
+    user = User.where(mobile: @mobile).first_or_create(@user_attributes.to_hash)
```

Comment:
I think it's probably better to move the `to_hash` up into the controller. The builder should expect a normal hash so that is can also be valid to call this code from elsewhere and not be tied to having controller params.